addons/account/tests/test_account_move_entry.py

```diff
@@ -530,6 +533,54 @@ def test_journal_sequence_ordering(self):
         self.assertEqual(copies[5].name, 'XMISC/2019/10005')
         self.assertEqual(copies[5].state, 'draft')
 
+    def test_sequence_concurency(self):
+        with db_connect(self.env.registry.db_name).cursor() as cr0,\
+                db_connect(self.env.registry.db_name).cursor() as cr1,\
+                db_connect(self.env.registry.db_name).cursor() as cr2:
+            env0 = api.Environment(cr0, SUPERUSER_ID, {})
+            env1 = api.Environment(cr1, SUPERUSER_ID, {})
+            env2 = api.Environment(cr2, SUPERUSER_ID, {})
+
+            journal = env0['account.journal'].create({
+                'name': 'concurency_test',
+                'code': 'CT',
+                'type': 'general',
+            })
+            account = env0['account.account'].create({
+                'code': 'CT',
+                'name': 'CT',
+                'user_type_id': env0.ref('account.data_account_type_fixed_assets').id,
+            })
+            move_ids = env0['account.move'].create([{
+                'journal_id': journal.id,
+                'date': fields.Date.from_string('2016-01-01'),
+                'line_ids': [(0, 0, {'name': 'name', 'account_id': account.id})]
+            } for i in range(3)])
+            move_ids.name = '/'
+            move_ids[0].post()
+            self.assertEqual(move_ids.mapped('name'), ['CT/2016/01/0001', '/', '/'])
+            env0.cr.commit()
+
+            # start the transactions here to simulate concurency
+            env1.cr.execute('SELECT 1 FROM account_move WHERE FALSE')
+            env2.cr.execute('SELECT 1 FROM account_move WHERE FALSE')
```

Comment:
I don't understand the purpose of those queries.  Please explain.

Reply:
They are there so that the transactions start on each cursor. The database will then treat the two cursors as concurrent, and the `UPDATE` done in `sequence.mixin` on the `account.move` with the highest number will fail on the second query (on `env2.cr`)

Reply:
"Make the transaction start" makes sense.  But then a plain stupid `'SELECT 1'` does the job, doesn't it?
I was confused, thinking that you were trying to have some READ LOCK on the table.